auth/src/user.rs

```diff
@@ -246,6 +275,32 @@ impl User {
 
         Ok(false)
     }
+
+    async fn add_subscription_items(
+        &self,
+        stripe_client: &stripe::Client,
+        subscription_items: stripe::UpdateSubscriptionItems,
+    ) -> Result<(), Error> {
+        // A pro or admin account tier is required to get to this point, but the ssubscription sync
+        // ahead of this call may lead to an account downgrade.
+        if !matches![self.account_tier, AccountTier::Pro | AccountTier::Admin] {
+            error!("account was downgraded from pro in sync, denying the addition of new items");
+            return Err(Error::Unauthorized);
+        }
+
+        let Some(ref subscription_id) = self.subscription_id else {
+            return Err(Error::MissingSubscriptionId);
+        };
```

Comment:
Since this is the data layer, it should not be doing business logic checks like these. Rather the handler should do these and this function should only update the needed "datastores"

Reply:
The difficulty with this is that the subscription is synced with stripe in the call before this, in `UserManager::add_subscription_items`, so the tier could have changed between this call and the handler checks. But I can investigate refactoring it so the subscription is synced earlier.

Reply:
Addressed in latest commits.